### PR TITLE
Simplified multi channel demos cmake

### PR DESCRIPTION
**Pull request by ilya-lavrenov**

### CODE REVIEW

File: demos/multi_channel/common/CMakeLists.txt [hunk 1]
```diff
@@ -94,22 +91,20 @@ if(MULTICHANNEL_DEMO_USE_NATIVE_CAM)
...
+find_package(Threads REQUIRED)
+
 target_link_libraries(${TARGET_NAME}
     PRIVATE ${InferenceEngine_LIBRARIES} gflags ${OpenCV_LIBRARIES}
-    PUBLIC common)
-
-if(UNIX)
-    target_link_libraries(${TARGET_NAME} PRIVATE pthread)
-endif()
+    PUBLIC Threads::Threads common)
```
Comment (Wovchena) [inline demos/multi_channel/common/CMakeLists.txt:110]: Can you explain to me why `multicam` and `Threads::Threads` are linked publicly now?

Comment (ilya-lavrenov) [reply demos/multi_channel/common/CMakeLists.txt:110]: Because they are used in headers.

Comment (Wovchena) [reply demos/multi_channel/common/CMakeLists.txt:110]: I see. I've tried compiling with `PRIVATE Threads::Threads` and it passed. That's why I asked. But `main.cpp`s actually have `#include <thread>` even though they don't need it. Probably should be removed later.

**Review by Wovchena — APPROVED** [review]

File: demos/multi_channel/common/CMakeLists.txt (hunk 1 above)
Comment (ilya-lavrenov) [reply demos/multi_channel/common/CMakeLists.txt:110]: have you tried to enable TBB for multi channel demo?

Comment (Wovchena) [reply demos/multi_channel/common/CMakeLists.txt:110]: Yes, passes for Windows with `-DMULTICHANNEL_DEMO_USE_TBB=ON` and `PRIVATE Threads::Threads`.

**Pull request merged**